src/services/lokapiService.ts

```diff
@@ -12,11 +12,34 @@ class LokAPI extends LokAPIBrowserAbstract {
     cyclos
   }
 
-  requestLocalPassword = async function (state: string) {
+  rememberedPassword = { pwd: '', inputTime: 0 }
+
+  requestLocalPassword = async (state: string) => {
     let text
+
+    // Lets see if we have a (still valid) password in there...
+    // ... that was used less than 30secs ago
+    // If it's the case, we will provide it to the lokapi right away
+    if (this.rememberedPassword.pwd.length > 0
+      && this.rememberedPassword.inputTime + 30000 > Date.now()
+      && state === 'firstTry'
+    ) {
+      // Valid pwd -> update the last used time and provide
+      // the pwd to the lokapi
+      this.rememberedPassword.inputTime = Date.now()
+      return this.rememberedPassword.pwd
+
+    // No valid pwd
+    // so we make sure the pwd is reset
+    // and ask for it again
+    } else {
+      this.rememberedPassword = { pwd: '', inputTime: 0 }
+    }
+
```

Comment:
You can't use this. You should probably use closures to keep an internal variables. I can help you.

Reply:
Voilà @vaab , latest commit (https://github.com/Lokavaluto/lokavaluto-web-application/pull/47/commits/28498a34f61ad4cc7e8a6bee46244855850207a1) makes the function a closure, is that correct in your view ?